app/controllers/api/databases.php

```diff
@@ -2535,20 +2535,19 @@ function updateAttribute(
             throw new Exception(Exception::COLLECTION_NOT_FOUND);
         }
 
-        $indexes = $collection->getAttribute('indexes');
-
-        // Search for index
-        $indexIndex = array_search($key, array_map(fn($idx) => $idx['key'], $indexes));
+        $index = $dbForProject->findOne('indexes', [
+            Query::equal('$id', [$database->getInternalId() . '_' . $collection->getInternalId() . '_' . $key])
+        ]);
```

Comment:
I don't think we should use `findOne()` because it incurs a db operation while the array_search is a local operation.

Reply:
removed this